modules/permission-management/src/Volo.Abp.PermissionManagement.Domain/Volo/Abp/PermissionManagement/PermissionManagementProvider.cs

```diff
@@ -23,18 +24,33 @@ public abstract class PermissionManagementProvider : IPermissionManagementProvid
             GuidGenerator = guidGenerator;
             CurrentTenant = currentTenant;
         }
-        
+
         public virtual async Task<PermissionValueProviderGrantInfo> CheckAsync(string name, string providerName, string providerKey)
         {
+            var multiplePermissionValueProviderGrantInfo = await CheckAsync(new[] {name}, providerName, providerKey);
+
+            return multiplePermissionValueProviderGrantInfo.Result.First().Value;
+        }
+
+        public virtual async Task<MultiplePermissionValueProviderGrantInfo> CheckAsync(string[] names, string providerName, string providerKey)
+        {
+            var multiplePermissionValueProviderGrantInfo = new MultiplePermissionValueProviderGrantInfo(names);
             if (providerName != Name)
             {
-                return PermissionValueProviderGrantInfo.NonGranted;
+                return multiplePermissionValueProviderGrantInfo;
             }
 
-            return new PermissionValueProviderGrantInfo(
-                await PermissionGrantRepository.FindAsync(name, providerName, providerKey) != null,
-                providerKey
-            );
+            var permissionGrants = await PermissionGrantRepository.GetListAsync(names, providerName, providerKey);
+
+            foreach (var permissionName in names)
+            {
+                if (permissionGrants.Any(x => x.Name == permissionName))
+                {
+                    multiplePermissionValueProviderGrantInfo.Result[permissionName] = new PermissionValueProviderGrantInfo(true, providerKey);
+                }
+            }
+
+            return multiplePermissionValueProviderGrantInfo;
```

Comment:
We also need add `providerKey` to `multiplePermissionValueProviderGrantInfo` even it's not granted.

```cs
return new PermissionValueProviderGrantInfo(
      await PermissionGrantRepository.FindAsync(name, providerName, providerKey) != null,
      providerKey
  );
```